foxpuppet/windows.py

```diff
@@ -0,0 +1,48 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from contextlib import contextmanager
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class Windows(object):
+
+    def __init__(self, selenium):
+        self.selenium = selenium
+        self._handle = None
+
+    @property
+    def all(self):
+        return self.selenium.window_handles
+
+    @property
+    def current(self):
+        return self.selenium.current_window_handle
+
+    def focus(self, handle):
+        return self.selenium.switch_to.window(handle)
+
+    def get_new_window_handle(self, handles):
+        """Get new window handle"""
+        if handles != self.selenium.window_handles:
+            new_handle = self.selenium.window_handles[len(handles):]
+        return new_handle
```

Comment:
Well this would only return the newest handle. Which means you would need to know if you need a specific handle when you create it. If this isn't correct, i'll have to adapt the Puppeteer setup.